gui/src/2D/controls/container.ts

```diff
@@ -516,6 +517,10 @@ export class Container extends Control {
             }
         }
 
+        if (!super.contains(x, y)) {

```

Comment:
instead of calling "contains" twice, store it in a variable and use the same variable twice. save a few CPU cycles :-)